third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -1286,6 +1287,18 @@ fn parse_for_loop(context: &mut Context) -> Result<(Exp, bool), Box<Diagnostic>>
         SequenceItem_::Bind(flag, None, Box::new(false_exp)),
     );
 
+    // To create the declaration "let ub_value = upper_bound", first create the variable flag, and
+    // then assign it to upper_bound
+    let ub_value_symbol = Symbol::from("__upper_bound_value");
```

Comment:
Don't use such a symbol, which may clash with user symbols.  Instead, be sure to include a special character (perhaps `$` or `#`) or fix https://github.com/aptos-labs/aptos-core/issues/11430 and use a shared method.

Reply:
Also, what happens with nested loops?  Shouldn't this symbol be unique for each loop?

Reply:
Symbol with `$` cannot be used at the parser level. I have tried some workarounds but to no success. I am planning to fix that issue separately. 

These variables are specific to the context of a given loop. We confirm that nested loops work as expected in the tests.